src/controllers/project.controller.ts

```diff
@@ -746,6 +738,47 @@ export class ProjectController {
 		}
 	}
 
+	async setProjectResourcesDirectory(resourcesPath: string): Promise<void> {
+		try {
+			const newDirectoryUri = Uri.file(resourcesPath);
+			const newLocalResourcesPath = await window.showOpenDialog({
+				canSelectFolders: true,
+				canSelectFiles: false,
+				defaultUri: newDirectoryUri,
+
+				openLabel: translate().t("projects.setResourcesDirectory"),
+			});
+
+			if (newLocalResourcesPath === undefined || newLocalResourcesPath.length === 0) {
+				return;
+			}
+			const savePath = newLocalResourcesPath[0].fsPath;
+
+			await commands.executeCommand(vsCommands.setContext, this.projectId, { path: savePath });
```

Comment:
Check if the path is the same, don't do extra actions.
consult with @haimzlato if we should display the success notification.

Reply:
refactored - after discussion - even if the paths are the same, we will display the notification, but we won't update the global context because the paths are the same.